File: cs470/proj1/Tennis/TieBreakerScore.cpp
// Changes made by Zachary Thompson

#include "TieBreakerScore.hpp"
#include <iostream>
#include <cstdlib>

TieBreakerScore::TieBreakerScore( Player *p1, Player *p2 ): Score(p1, p2) {}

bool TieBreakerScore::haveAWinner() {
    return ( p1Score >= 7 || p2Score >= 7 ) && abs( p1Score - p2Score ) >= 2;
}

void TieBreakerScore::print() {
    std::cout << "  (tie breaker  " << player1Score() << "-" << player2Score() << ")";
}

File: cs470/proj1/ObjectiveCTennis/TennisInObjectiveC/TieBreakerScore.m
//  Additions made by Zachary Thompson
//  TieBreakerScore.m
//  TennisInObjectiveC
//
//  Created by student on 2/3/14.
//  Copyright (c) 2014 Ali Kooshesh. All rights reserved.
//

#import "TieBreakerScore.h"

@implementation TieBreakerScore

-(BOOL) haveAWinner
{
    return ( self.player1Score >= 7 || self.player2Score >= 7) && abs(self.player1Score - self.player2Score) >= 2;
}

-(NSString *) description
{
    return [NSString stringWithFormat:@"  (tie breaker %d-%d)", self.player1Score, self.player2Score];
}

@end
